sycl/include/CL/sycl/detail/pi.h

```diff
@@ -1488,6 +1489,8 @@ __SYCL_EXPORT pi_result piSamplerRelease(pi_sampler sampler);
 //
 // Queue Commands
 //
+__SYCL_EXPORT pi_result piKernelLaunch(pi_queue queue);
```

Comment:
I have an idea for an alternative approach here that doesn't involve adding a new PI entry-point, we use `piQueueFlush` instead. As it already exists in PI with the semantics of starting execution of work lazily scheduled, assuming it works the same as [clFlush](https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_API.html#clFlush). I'd then expect a flush to happen on event wait as well as queue wait.

`piQueueFlush`  is also more generic than just for kernel execution commands. So if in the future we wanted more than kernel launch commands to be lazily executed, we wouldn't need to add another entry-point. e.g. could lazily enqueue `piEnqueueMemBufferCopy` commands and flush them. Rather than having to add a `piMemBufferCopy` entry-point to match the `piEnqueueMemBufferCopy` like we've done here with `piEnqueueKernelLaunch`.